docs/local-module.md

```diff
@@ -0,0 +1,60 @@
+# Postman.gov.sg
+
+- [Background](#background)
+- [How to install](#how-to-install)
+- [Local development](#local-development)
+- [Build dependencies](#build-dependencies)
+- [More information](#more-information)
+
+## Background
+
+On Postman, we made use of a local node module to share code between `frontend`, `backend` & `worker`. It is similar to how one would install npm modules from npm registry or github, except that the path is pointed to a local folder.
+
+ie: `"postman-templating": "file:../modules/postman-templating"`.
+
+We wanted to find a way to share the templating logic because:
+
+1. Used in frontend, backend & worker.
+2. Better consistency, we just have to change the code in one place rather than three.
+
+There are several options that we considered:
+
+- npm install from separate repo (postmangovsg-js-tools?)
+- npm pack and copy tarball into respective folders
+- Publish to npm registry
+- Separate branch (sdk branch)
+- Use lerna to manage multi-package repo
+- local node module
+
+We decided to use local shared module as the rest of the options are more complicated or requires alot more time & effort to set them up.
+
+## How to install
+
+Any shared module can be installed using `npm install ../modules/postman-templating`.
+
+## Local development
+
+To test the changes you want to make to a module like `postman-templating`:
+
+1. Make changes to the module.
+2. Run `npm start build` in `postman-templating`.
+
+`npm start build` will remove & replace the `build` folder in `postman-templating` with the updated change.
+
+## Deploying or building in Travis
+
+If there are any new dependencies installed in the local node module, the root folder's `package-lock.json` needs to be updated. This can be done through deleting the `node_modules` and running `npm install`.
```

Comment:
Change "root" to "parent" for clarity. Root implies the top level.

Reply:
Good point!